### PR TITLE
Workload constraints (step 1)

### PR DESCRIPTION
**Pull request by jewertow**
PR adapts `SelectiveWorkBalancer` for use workload constraints.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -70,6 +70,7 @@ allprojects {
         compile group: 'org.slf4j', name: 'slf4j-api', version: '1.7.7'
 
         testCompile group: 'junit', name: 'junit', version: '4.11'
+        testCompile group: 'com.tngtech.java', name: 'junit-dataprovider', version: '1.10.0'
```
Comment (mkopylec) [inline build.gradle]: Why is this dep added?

Comment (jewertow) [reply build.gradle]: I needed parameterized test. (https://github.com/allegro/hermes/pull/1083/files#diff-d6b07aceab798e7a80aff1c6ca7e420dR253)

File: hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/workload/constraints/WorkloadConstraints.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                               int availableConsumers) {
+        this.subscriptionConstraints = subscriptionConstraints != null ? subscriptionConstraints : emptyMap();
+        this.topicConstraints = topicConstraints != null ? topicConstraints : emptyMap();
+        this.consumersPerSubscription = consumersPerSubscription;
+        this.maxSubscriptionsPerConsumer = maxSubscriptionsPerConsumer;
+        this.availableConsumers = availableConsumers;
+    }
+
+    public int getConsumersNumber(SubscriptionName subscriptionName) {
+        final int requiredConsumers = getSubscriptionConstraints(subscriptionName)
```
Comment (dankraw) [inline hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/workload/constraints/WorkloadConstraints.java]: the algorithm could be simplified so have no code duplication:

```
final int requiredConsumers = subscriptionConstraints.getOrDefault(subscriptionName, topicConstraints.get(subscriptionName.getTopicName()))
.map(Constraints::getConsumersNumber)
.orElse(0);
```

Comment (jewertow) [reply hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/workload/constraints/WorkloadConstraints.java]: Fixed

**Pull request merged**